src/FixedRankMatrices.jl

```diff
@@ -0,0 +1,177 @@
+using LinearAlgebra: diag, Diagonal, svd, SVD, rank, dot
+
+@doc doc"""
```

Comment:
This manifold is missing many methods that I took for granted previously, some tests will have to be made optional.

Reply:
That's why I did not directly do the tests, exp/log have to get optional for example.

Reply:
I found a way: I generalized all appearances of `log` (despite those that really test exp/log identity) to `inverse_retract` with a `default_inverse_retraction` that as a parameter defaults to `LogarithmicInverseRetraction`as a parameter. Now I just have to fix my inverse retractions, still.